src/lib.rs

```diff
@@ -106,6 +107,33 @@ where
         }
     }
 
+    /// Returns a vector of current counts of token T
+    fn rank<S: AsRef<[T]>>(&self, tokens: S) -> Vec<(&Token<T>, &usize)> {
+        let tokens = tokens.as_ref();
+        if tokens.is_empty() {
+            return Vec::new();
+        }
+        let mut toks = vec![];
+        toks.extend(tokens.iter().map(|token| Some(token.clone())));
+        if !self.map.contains_key(&toks) {
+            return Vec::new();
+        }
+        println!("Tokens {:?}", toks);
+        println!("Map {:?}", self.map);
+        let result = self.map.get(&toks).unwrap();
+        let sorted: Vec<_> = result
+            .iter()
+            .sorted_by(|&a, &b| Ord::cmp(a.1, b.1).reverse())
+            .collect();
+        sorted
+    }
+
+    /// Get iterator over all tokens following a given set of tokens
+    /// (sorted by count)
+    pub fn iter_rank<S: AsRef<[T]>>(&self, tokens: S) -> RankIterator<T> {
+        RankIterator::new(self, tokens)
+    }
```

Comment:
Add feature to specify a token to be searched for and return its rank eg. iterate over all tokens of token `a` as predecessor and search for token `b` and return its rank